server/src/main/java/org/opensearch/search/aggregations/metrics/AvgAggregator.java

```diff
@@ -126,6 +147,59 @@ public void collect(int doc, long bucket) throws IOException {
         };
     }
 
+    public LeafBucketCollector getStarTreeLeafCollector(LeafReaderContext ctx, LeafBucketCollector sub, CompositeIndexFieldInfo starTree)
+        throws IOException {
+        StarTreeValues starTreeValues = StarTreeQueryHelper.getStarTreeValues(ctx, starTree);
+        assert starTreeValues != null;
+
+        String fieldName = ((ValuesSource.Numeric.FieldData) valuesSource).getIndexFieldName();
+        String sumMetricName = StarTreeUtils.fullyQualifiedFieldNameForStarTreeMetricsDocValues(
+            starTree.getField(),
+            fieldName,
+            MetricStat.SUM.getTypeName()
+        );
+        String countMetricName = StarTreeUtils.fullyQualifiedFieldNameForStarTreeMetricsDocValues(
+            starTree.getField(),
+            fieldName,
+            MetricStat.VALUE_COUNT.getTypeName()
+        );
+
+        final CompensatedSum kahanSummation = new CompensatedSum(sums.get(0), 0);
+        SortedNumericStarTreeValuesIterator sumValuesIterator = (SortedNumericStarTreeValuesIterator) starTreeValues
+            .getMetricValuesIterator(sumMetricName);
+        SortedNumericStarTreeValuesIterator countValueIterator = (SortedNumericStarTreeValuesIterator) starTreeValues
+            .getMetricValuesIterator(countMetricName);
+        FixedBitSet matchedDocIds = getStarTreeFilteredValues(context, ctx, starTreeValues);
+        assert matchedDocIds != null;
+
+        int numBits = matchedDocIds.length();  // Get the length of the FixedBitSet
+        if (numBits > 0) {
+            // Iterate over the FixedBitSet
+            for (int bit = matchedDocIds.nextSetBit(0); bit != -1; bit = bit + 1 < numBits ? matchedDocIds.nextSetBit(bit + 1) : -1) {
+                // Advance to the bit (entryId) in the valuesIterator
+                if (sumValuesIterator.advance(bit) == StarTreeValuesIterator.NO_MORE_ENTRIES
```

Comment:
Lets change from advance to advanceExact in all aggregators.  ( Ignore if already this change is being made )